src/Tools/Pagination/Paginator.php

```diff
@@ -228,7 +228,22 @@ private function appendTreeWalker(Query $query, string $walkerClass): void
      */
     private function getCountQuery(): Query
     {
-        $countQuery = $this->cloneQuery($this->query);
+        /*
+            As opposed to using self::cloneQuery, the following code does not transfer
+            a potentially existing result set mapping (either set directly by the user,
+            or taken from the parser result from a previous invocation of Query::parse())
+            to the new query object. This is fine, since we are going to completely change the
+            select clause anyway, so a previously existing result set mapping (RSM) is probably wrong anyway.
```

Comment:
Let's avoid repeating "anyway"?

```suggestion
            select clause, so a previously existing result set mapping (RSM) is probably wrong anyway.
```